doc/rl-2505: wrangle in nixpkgs and nixos split

getchoo opened the pull request:
Slight corrections needed since the notes first split

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

getchoo committed:
doc/rl-2505: move some entires from nixos notes

These are more broadly applicable to Nixpkgs than only NixOS

getchoo committed:
doc/rl-2505: de-duplicate nixos entires

Make sure package updates are reported in the Nixpkgs notes

getchoo committed:
nixos/doc/rl-2505: use "Release" as header name

This keeps us in-line with all previous releases. Users should be able
to still differentiate the NixOS and Nixpkgs release notes based on the
header of Nixpkgs' notes

leona-ya commented on doc/release-notes/rl-2505.section.md:
```diff
@@ -272,27 +294,6 @@
 
 - The default version of `z3` has been updated from 4.8 to 4.14, and all old versions have been dropped. Note that `fstar` still depends on specific versions, and maintains them as overrides.
 
-- `prometheus` has been updated from 2.55.0 to 3.1.0.
```
I think these should be in the Nixpkgs docs instead of the NixOS docs as they describe the package. At least this is my understanding of the doc-split...

fricklerhandwerk replied:
Yes. That we're handling derivations and service configuration separately is in part a feature and in part a huge UX curse (for which there are possible ways out), but the idea is to at least make that explicit.

getchoo committed:
doc/rl-2505: move all package entries from nixos notes

This ensures the NixOS notes only refer to things like modules and
NixOS-specific improvements, while the Nixpkgs notes refer to all
package changes

fricklerhandwerk commented on doc/release-notes/rl-2505.section.md:
```diff
@@ -386,14 +450,70 @@
+  This allows packaging configuration to be overridden in a consistent manner by
+  providing an alternative to `rec {}` syntax.
+
+- Caddy can now be built with plugins by using `caddy.withPlugins`, a `passthru` function that accepts an attribute set as a parameter. The `plugins` argument represents a list of Caddy plugins, with each Caddy plugin being a versioned module. The `hash` argument represents the `vendorHash` of the resulting Caddy source code with the plugins added.
```
Hm now this should be in NixOS, no?

getchoo replied:
`caddy.withPlugins` is a Nixpkgs interface. I think the module excerpt here is just a way to illustrate it rather than say it's only for NixOS

fricklerhandwerk replied:
Ah, but then let's just remove the module stuff around it, it would be super confusing in the Nixpkgs context.

fricklerhandwerk commented on doc/release-notes/rl-2505.section.md:
````diff
@@ -386,14 +450,70 @@
+
+  Example:
+  ```nix
+  {
+    services.caddy = {
+      enable = true;
+      package = pkgs.caddy.withPlugins {
+        plugins = [
+          # tagged upstream
+          "github.com/caddy-dns/powerdns@v1.0.1"
+          # pseudo-version number generated by Go
+          "github.com/caddy-dns/cloudflare@v0.0.0-20240703190432-89f16b99c18e"
+          "github.com/mholt/caddy-webdav@v0.0.0-20241008162340-42168ba04c9d"
+        ];
+        hash = "sha256-wqXSd1Ep9TVpQi570TTb96LwzNYvWL5EBJXMJfYWCAk=";
+      };
+    };
+  }
````
```suggestion
  let pkgs = import <nixpkgs> { }; in
  pkgs.caddy.withPlugins {
      plugins = [
        # tagged upstream
        "github.com/caddy-dns/powerdns@v1.0.1"
        # pseudo-version number generated by Go
        "github.com/caddy-dns/cloudflare@v0.0.0-20240703190432-89f16b99c18e"
        "github.com/mholt/caddy-webdav@v0.0.0-20241008162340-42168ba04c9d"
      ];
      hash = "sha256-wqXSd1Ep9TVpQi570TTb96LwzNYvWL5EBJXMJfYWCAk=";
    };
  }
```

getchoo committed:
doc/rl-2505: re-introduce note about __structuredAttrs & python

Seems this was accidentally removed in 8ae9309 instead of only being
moved from the highlight section
